processing/src/main/java/io/druid/query/metadata/SegmentMetadataQueryQueryToolChest.java

```diff
@@ -181,9 +181,11 @@ public SegmentAnalysis apply(SegmentAnalysis arg1, SegmentAnalysis arg2)
       public byte[] computeCacheKey(SegmentMetadataQuery query)
       {
         byte[] includerBytes = query.getToInclude().getCacheKey();
-        return ByteBuffer.allocate(1 + includerBytes.length)
+        byte[] analysisTypesBytes = query.getAnalysisTypesCacheKey();
```

Comment:
a bit pedantic, but if there were enough analysis types that they could conceivably serialize to a valid column name, it's possible for the caching to get confused for a query with a ListColumnIncluderator. Some kind of separator should help with that.